tests/openassetio/hostAPI/test_manager.py

```diff
@@ -570,14 +585,16 @@ def test_wraps_the_corresponding_method_of_the_held_interface(
             assert manager.getRelatedReferences(
                 refs_arg, specs_arg, a_context) == method.return_value
             method.assert_called_once_with(
-                expected_refs_arg, expected_specs_arg, a_context, host_session, resultSpec=None)
+                expected_refs_arg, expected_specs_arg, a_context, host_session,
+                resultTraitSet=None)
             method.reset_mock()
 
-        # Check optional resultSpec
+        # Check optional resultTraitSet
```

Comment:
[minor] Do we need to include the type of the collection in the name? maybe it can just be resultTraits to make it shorter to type/read.

Reply:
@feltech and I debated this when we worked on #301. We wanted to avoid the ambiguity of just saying "traits" for two reasons:

 - In many cases, this argument is present in a batched call, where an argument is a list of trait sets. As traits is already plural, it causes confusion (eg: [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/297/files#diff-0e6e2f71c0e91d80fc6b3ffa29832f79a53f4372224bafc9336b03859a7648e7R459). It felt like saying 'traitSets' was clear that it was multiple trait sets, rather than "traits" which seemed more readily interpreted as multiple trait IDs rather than multiple sets of ids. We need to be consistent in naming across all parameters that take this type, and so it becomes particularly relevant in this specific case - as it makes it clear that this argument is constant across the batch. If it said "resultTraits" it could be construed as a one per batch element (as `entityRefs` and `relationshipSpecs` are).

 - Given that traits can be instantiated, and this only refers to a set of IDs. Colloqually, for #249 and other documentation updates around the unified data model we wanted to work on a coherent use of language, we regularly use "the set of traits" in that documentation, to the point where a "trait set" almost becomes a proper noun for a collection of trait IDs.  As such, we tried to use that in all the places where a method takes a set of IDs (rather than a specification with data for these traits) for consistency , to try make it not sound like "some multiple trait instances". It's more tenuous than the first reason, but hopefully reinforceable by the documentation.

Reply:
Sounds like fair reasoning.